merlin/systems/dag/ops/fil.py

```diff
@@ -90,7 +87,20 @@ def compute_input_schema(
         """Return the input schema representing the input columns this operator expects to use."""
         return self.input_schema
 
-    def export(self, path, input_schema, output_schema, params=None, node_id=None, version=1):
+    @property
+    def exportable_backends(self):
+        return ["ensemble", "executor"]
+
+    def export(
+        self,
+        path: str,
+        input_schema: Schema,
+        output_schema: Schema,
+        params: dict = None,
+        node_id: int = None,
+        version: int = 1,
+        backend: str = "ensemble",
```

Comment:
This `backend` doesn't appear to be used in this method. I'm guessing it's here for consistency of export signatures?

Reply:
However, There is one thing confusing me about this `backend` parameter. It's that the value of this `backend` for the `PipelineableInferenceOperator` export method means something different. In that method the backend  corresponds to the backend that will be written out in the Triton config. Which suggest that the `ensemble` is only valid for the `Ensemble` and not for all operators.

Reply:
> the backend corresponds to the backend that will be written out in the Triton config

This is always true, but there's a wobble in the logic here. An operator that isn't exportable for a given format (e.g. `ensemble`) won't have its `export` method called, but some operators need to write artifacts to disk in cases where they won't write out a Triton model config. In those cases, we currently consider them exportable for the `ensemble` back-end, but return an empty model config from the `export` method.

We could have resolved these weird inconsistencies by introducing back-ends/run-times/adapters in this PR (since then each back-end could have its own implementation of `export` for any given operator via the operator implementation table), but we wrote it this way in part to motivate the refactor in that direction.

Reply:
In the context of this `PredictForest` operator, which currently has `exportable_backends = ["ensemble", "executor"]`. Based on what you described I would assume that for operators that include `executor` in `exportable_backends` then we would need to perform a conditional logic in the `export` method.

Reply:
If I'm understanding this correctly. I think what this means is that we'll end up having unused models written out to the model repository (if the op contains `executor` in `exportable_backends`, then we'll be calling the transform method  as part of the LocalExecutor), but we'll also be exporting the node in the same way as for the ensemble case (if there's no conditional logic similar to the Faiss operator). I suppose that doesn't break any functionality, but maybe not the intention?

Reply:
> for operators that include executor in exportable_backends then we would need to perform a conditional logic in the export method
It depends on how the operator is structured. For operators that run in custom Triton back-ends, we still need to export a separate model directory. For operators that wrap one operator in another (e.g. in order to align the schema of the outer operator with the graph), the outer wrapper op runs in the `Executor` (in plain Python) but the inner op runs in a custom back-end.

Reply:
> I suppose that doesn't break any functionality (since, if the op contains executor in exportable_backends, then we'll be calling the transform method as part of the LocalExecutor), but maybe not the intention?

I think this is a case where we're using the operator's ensemble export twice:
* Once in the actual export of the full ensemble to run as a Triton ensemble
* Once for exporting the single operator as a separate model directory to be used in `Executor` mode (still required to use specific non-Python Triton back-ends)